packages/react/src/lib/createVariant.ts

```diff
@@ -0,0 +1,38 @@
+import hoistNonReactStatic from 'hoist-non-react-statics'
+
+function createVariant<TProps>(
```

Comment:
make API consistent with the one provided by `createComponent` function - at the end it should match the following (prop names may be different):

```jsx
const Variant = createVariant({
  forComponent: Input,
  withVariantName: 'Minimalistic',
  withPropTypes: ...,
  withDefaultProps: ...
})
```

and preserve the same type safety checks it has now.